array.h

```diff
@@ -881,6 +881,11 @@ NDARRAY_HOST_DEVICE T convert_dims(const U& u, internal::index_sequence<Is...>)
   return std::make_tuple(convert_dim<Is, T>(u)...);
 }
 
+template <size_t DstRank, class U, size_t... Is>
+NDARRAY_HOST_DEVICE bool is_trivial_slice(const U& u, internal::index_sequence<Is...>) {
+  return all((Is < DstRank || std::get<Is>(u).extent() == 1)...);
```

Comment:
Ugh this is hard for me to exactly wrap my head around. Add a comment?

Say what `u` is first.

Then, it's something like - return true if for each index i, if i >= u.rank, then extent(i) must be 1?